src/lib/y2storage/encryption_processes/apqn.rb

```diff
@@ -0,0 +1,147 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast2/execute"
+
+module Y2Storage
+  module EncryptionProcesses
+    # Class representing an APQN used for generating a new secure key, see {SecureKey}.
+    #
+    # For more information, see
+    # https://www.ibm.com/support/knowledgecenter/linuxonibm/com.ibm.linux.z.lxdc/lxdc_zkey_reference.html
+    class Apqn
+      # Location of the lszcrypt command
+      LSZCRYPT = "/sbin/lszcrypt".freeze
+      private_constant :LSZCRYPT
+
+      class << self
+        # All APQNs found in the system
+        #
+        # @return [Array<Apqn>]
+        def all
+          create_apqns
+        end
+
+        # All online APQNs found in the system
+        #
+        # @return [Array<Apqn>]
+        def online
+          all.select(&:online?)
+        end
+
+        private
+
+        # Creates APQNs from the data read in the system
+        #
+        # @return [Array<Apqn>]
+        def create_apqns
+          apqns_data.map { |d| new(*d) }
+        end
+
+        # Data of the APQNs found in the system
+        #
+        # For each APQN, it returns its name (card.domain), type, mode, status and requests, see
+        # {#execute_lszcrypt}.
+        #
+        # @return [Array<Array<String>>]
+        def apqns_data
+          data = execute_lszcrypt.split("\n").map(&:split)
+          return [] if data.size == 2
+
+          # remove header
+          data.shift(2)
+
+          # select card.domain entries
+          data.select { |d| d.first.match?(/.*\..*/) }
```

Comment:
Are `.01` or `01.` valid names? If you always need something before and after the point, you could consider using `.+` instead of `.*`. If it does not matter, I would simply use `d.first.include?(".")`.

Reply:
Yes, `.+` is the correct regexp here.